core/sail/shacl/src/main/java/org/eclipse/rdf4j/sail/shacl/ShaclSail.java

```diff
@@ -768,6 +746,12 @@ public void setPerformanceLogging(boolean performanceLogging) {
 	 * @return <code>true</code> if serializable validation is enabled, <code>false</code> otherwise.
 	 */
 	public boolean isSerializableValidation() {
+		if (getBaseSail() instanceof SchemaCachingRDFSInferencer) {
+			if (serializableValidation) {
+				logger.error("SchemaCachingRDFSInferencer is not supported when using serializable validation!");
```

Comment:
Given that we're not actually stopping execution, I think this should be a warning rather than an error.